web/src/patches/utils.ts

```diff
@@ -0,0 +1,182 @@
+import { Address, Slot } from '@stencila/stencila'
+import GraphemeSplitter from 'grapheme-splitter'
+
+/**
+ * Panic if there is a conflict between a `DomPatch` and the current DOM.
+ *
+ * This module make liberal use of assertions of consistency between `DomOperation`s
+ * and the current DOM with the view that if there is any inconsistency detected then
+ * it is best to simply exit the `applyPatch` function early and reload the page.
+ *
+ * This should only happen if there (a) the client has missed a `DomPatch`
+ * such that the state of the DOM is out of sync with the server-side document, or
+ * (b) if there is a bug in the following code. Hopefully testing rules out (b).
+ *
+ * Reloads the document to get a new DOM state and then throws an exception for
+ * early exit from the calling function.
+ */
+export function panic(message: string): Error {
+  // TODO reload the document
+  return new Error(message)
```

Comment:
```suggestion
  location.reload();
```

Don't think it applies in our current use cases, but we might need to be considerate of  `same-origin` policies for.

> The reload may be blocked and a SECURITY_ERROR DOMException thrown. This happens if the origin of the script calling location.reload() differs from the origin of the page that owns the Location object. See Same-origin policy for more information.
> from MDN https://developer.mozilla.org/en-US/docs/Web/API/Location/reload

Reply:
Yeah, I had `location.reload()` earlier but removed it because it broke the tests (`jsdom` does not support it?) but also because I think we can do better than reload the entire page i.e. ask for a new complete HTML version of the document and replace the `slot="root"`